train_segmentation.py

```diff
@@ -197,23 +243,28 @@ def train(train_loader,
     :param batch_size: number of samples to process simultaneously
     :param ep_idx: epoch index (for hypertrainer log)
     :param progress_log: progress log file (for hypertrainer log)
-    :param vis_params: (dict) Parameters found in the yaml config file. Named vis_params because they are only used for
-                        visualization functions.
+    :param vis_batch_range: range of batches to perform visualization on. see README.md for more info.
+    :param vis_at_train: (bool) if True, will perform visualization at train time, as long as vis_batch_range is valid
     :param device: device used by pytorch (cpu ou cuda)
     :param debug: (bool) Debug mode
     :return: Updated training loss
     """
     model.train()
     train_metrics = create_metrics_dict(num_classes)
-    vis_at_train = get_key_def('vis_at_train', vis_params['visualization'], False)
-    vis_batch_range = get_key_def('vis_batch_range', vis_params['visualization'], None)
 
     with tqdm(train_loader, desc=f'Iterating train batches with {device.type}') as _tqdm:
         for batch_index, data in enumerate(_tqdm):
             progress_log.open('a', buffering=1).write(tsv_line(ep_idx, 'trn', batch_index, len(train_loader), time.time()))
 
-            inputs = data['sat_img'].to(device)
-            labels = data['map_img'].to(device)
+            try:  # For HPC when device 0 not available. Error: RuntimeError: CUDA error: invalid device ordinal
```

Comment:
why? the device type is already inferred and it's being passed into this function. should we be changing that in this try and except statement?

Reply:
Often, on HPC, our get_device_ids returns that gpu 2 and 3 are available but I get a runtime error when using those values instead of 0, 1. Maybe the fact that the machine it's being executed is a virtual machine. Not sure exactly why, but I got into this problem again recently since the ".to(device)" is used in multiple place in our code now.